src/core/stdcpp/exception.d

```diff
@@ -11,95 +11,109 @@
-            this(const bad_exception) nothrow { }
-            //bad_exception operator=(const bad_exception) nothrow { return this; }
-            //virtual ~this() nothrow;
-            override const(char)* what() const nothrow;
-        }
+        this() nothrow { }
+        this(const bad_exception) nothrow { }
+        //bad_exception operator=(const bad_exception) nothrow { return this; }
+        //virtual ~this() nothrow;
+        override const(char)* what() const nothrow;
     }
 }
-else version (CRuntime_Glibc)
+else version (CppRuntime_Gcc)
 {
-    extern (C++, std)
-    {
-        alias void function() unexpected_handler;
-        unexpected_handler set_unexpected(unexpected_handler f) nothrow;
-        void unexpected();
+    alias void function() unexpected_handler;
+    unexpected_handler set_unexpected(unexpected_handler f) nothrow;
+    void unexpected();
 
-        alias void function() terminate_handler;
-        terminate_handler set_terminate(terminate_handler f) nothrow;
-        void terminate();
+    alias void function() terminate_handler;
+    terminate_handler set_terminate(terminate_handler f) nothrow;
+    void terminate();
 
-        pure bool uncaught_exception();
+    pure bool uncaught_exception();
 
-        class exception
-        {
-            this();
-            //virtual ~this();
-            void dtor1();
-            void dtor2();
-            const(char)* what() const;
-        }
-
-        class bad_exception : exception
-        {
-            this();
-            //virtual ~this();
-            override const(char)* what() const;
-        }
+    class exception
+    {
+        this();
+        //virtual ~this();
+        void dtor1();
+        void dtor2();
+        const(char)* what() const;
+    }
+
+    class bad_exception : exception
+    {
+        this();
+        //virtual ~this();
+        override const(char)* what() const;
     }
 }
-else version (CRuntime_Microsoft)
+else version (CppRuntime_Microsoft)
 {
-    extern (C++, std)
+    class exception
     {
-        class exception
-        {
-            this();
-            this(const exception);
-            //exception operator=(const exception) { return this; }
+        this();
+        this(const exception);
+        //exception operator=(const exception) { return this; }
             //virtual ~this();
-            void dtor() { }
-            const(char)* what() const;
+        void dtor() { }
+        const(char)* what() const;
 
-          private:
-            const(char)* mywhat;
-            bool dofree;
-        }
+    private:
+        const(char)* mywhat;
+        bool dofree;
+    }
 
-        class bad_exception : exception
-        {
-            this(const(char)* msg = "bad exception");
-            //virtual ~this();
-        }
+    class bad_exception : exception
+    {
+        this(const(char)* msg = "bad exception");
+        //virtual ~this();
+    }
+}
+else version (CppRuntime_Clang)
```

Comment:
Not sure if `CppRuntime_Clang` is the best name since Clang can use both libc++ and stdc++.

Reply:
We could remove those and use `__traits(getTargetInfo, "something")`...

Reply:
@jacob-carlborg : Currently compiling on OSX == `CppRuntime_Clang` defined.
Also linking with libstdc++ is deprecated.

Reply:
I’m pretty sure it’s only deprecated on macOS and Clang works on other platforms as well.

Reply:
Let's not bike-shed this... I can't move forwards on anything until C++ unit-tests work >_<

Reply:
Fine.